core/src/agent/state.rs

```diff
@@ -96,14 +98,18 @@ fn reduce_commit_entry(
     action_wrapper: &ActionWrapper,
 ) {
     let action = action_wrapper.action();
-    let entry = unwrap_to!(action => Action::Commit);
+    let (entry_type, entry) = match action {
+        Action::Commit(entry_type, entry) => (entry_type, entry),
+        _ => unreachable!(),
+    };
 
     // @TODO validation dispatch should go here rather than upstream in invoke_commit
     // @see https://github.com/holochain/holochain-rust/issues/256
 
-    let res = state.chain.push_entry(&entry);
+    // @TODO assumes entry already added to CAS
```

Comment:
Shouldn't you also push entry in CAS here?
When/where would you expect the entry to be pushed in CAS otherwise?